actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -75,10 +75,12 @@ def formats
 
       # Sets the \variant for template.
       def variant=(variant)
-        if variant.is_a?(Symbol)
-          @variant = [variant]
-        elsif variant.nil? || variant.is_a?(Array) && variant.any? && variant.all?{ |v| v.is_a?(Symbol) }
+        if variant.nil?
           @variant = variant
```

Comment:
Just continuing my own ping-pong. It could also be:

```ruby
@variant = VariantInquirer.new Array(variant).tap do |variants|
  unless variants.all? { |v| v.is_a?(Symbol) }
    raise ArgumentError, "super long message"
  end
end
```